### DISCUSSION TITLE
How to change a custom workflow environment variable in a job step

### DISCUSSION THREAD

**Discussion by boxerab** [community/community#25266] [Actions]
<p>I set an environment variable for my workspace like so :</p>
<pre><code class="lang-auto">env:
  DATA_BRANCH: master
</code></pre>
<p>and then I would like to change it in one of the job steps (for macos)</p>
<p><code>echo {DATA_BRANCH=osx} &gt;&gt; $GITHUB_ENV</code></p>
<p>but this does not change the variable.</p>
<p>My complete yml file is:</p>
<aside class="onebox githubblob" data-onebox-src="https://github.com/GrokImageCompression/grok/blob/1e2bd0787a049722481cd9f8c7f4d220b8be4faa/.github/workflows/build.yml">
  <header class="source">

      <a href="https://github.com/GrokImageCompression/grok/blob/1e2bd0787a049722481cd9f8c7f4d220b8be4faa/.github/workflows/build.yml" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <h4><a href="https://github.com/GrokImageCompression/grok/blob/1e2bd0787a049722481cd9f8c7f4d220b8be4faa/.github/workflows/build.yml" target="_blank" rel="noopener nofollow ugc">GrokImageCompression/grok/blob/1e2bd0787a049722481cd9f8c7f4d220b8be4faa/.github/workflows/build.yml</a></h4>

      <pre><code class="lang-yml">name: Build

on: [push,pull_request,workflow_dispatch]

env:
  BUILD_TYPE: Debug
  DATA_ROOT: ${{ github.workspace }}/data
  DATA_BRANCH: master
  GENERATOR_PLATFORM:
  LD_LIBRARY_PATH: ${{ github.workspace }}/kdu

jobs:
  build:
    strategy:
       matrix:
        os: [ubuntu-latest, windows-latest, macos-latest]
    # The CMake configure and build commands are platform agnostic and should work equally
    # well on Windows or Mac.  You can convert this to a matrix build if you need
    # cross-platform coverage.
    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
</code></pre>

  This file has been truncated. <a href="https://github.com/GrokImageCompression/grok/blob/1e2bd0787a049722481cd9f8c7f4d220b8be4faa/.github/workflows/build.yml" target="_blank" rel="noopener nofollow ugc">show original</a>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="boxerab" data-post="1" data-topic="234285">
<div class="title">
<div class="quote-controls"></div>
 boxerab:</div>
<blockquote>
<p><code>echo {DATA_BRANCH=osx} &gt;&gt; $GITHUB_ENV</code></p>
</blockquote>
</aside>
<p>The format for writing to the environment file is incorrect, it should be:</p>
<pre><code class="lang-auto">echo DATA_BRANCH=osx &gt;&gt; $GITHUB_ENV
</code></pre>

**Comment by boxerab** [comment]
<aside class="quote group-Advisory-Council" data-username="airtower-luna" data-post="2" data-topic="234285">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p><code>echo DATA_BRANCH=osx &gt;&gt; $GITHUB_ENV</code></p>
</blockquote>
</aside>
<p>Thank you very much!</p>

**Comment by boxerab** [comment]
<p>The documentation is confusing, as it has curly braces, which <strong>are</strong> used in other cases.<br>
In those cases it is preceded by a dollar sign.</p>
